packit/api.py

```diff
@@ -874,8 +874,11 @@ def _run_kinit(self) -> None:
 
     def clean(self):
         """ clean up stuff once all the work is done """
-        # command handlers have nothing to clean
-        logger.debug("PackitAPI.cleanup (there are no objects to clean)")
+        # this is called in p-s: Handler.clean
+        if self.up._command_handler:
```

Comment:
I can imagine istead fo checking `self.up._command_handler`
use `@property` and `@property_setter` and when use `self.up.command_handler = None` --> call this cleanup

Reply:
I'm sorry I don't follow: that's exactly what's happening here, check if handler was initialized and if it was, clean it up

if we accessed self.up.command_handler directly, we would actually initialize it which would start deploying kubernetes pods only to clean them up right away

I agree it would be more straightforward to rename `_command_handler` to not imply privateness but that would be a massive change in the codebase which I wanted to avoid

Reply:
now that I think about your suggestion, it's about adding `@property.setter` which is the same solution to the problem but different implementation - I can take a look

Reply:
yes, exactly, same solution, just hiding this magic with `_*` and `*` what was not nice from my PoV

Reply:
I added a new method to check if _command_handler is initialized so we don't touch the private property from the outside